examples/equivalent_sources/gradient_boosted.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2018 The Harmonica Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# This code is part of the Fatiando a Terra project (https://www.fatiando.org)
+#
+"""
+Gradient-boosted equivalent sources
+===================================
+
+When trying to grid a very large dataset, the regular
+:class:`harmonica.EquivalentSources` might not be the best option: they will
+require a lot of memory for storing the Jacobian matrices involved in the
+fitting process of the source coefficients. Instead, we can make use of the
+gradient-boosted equivalent sources, introduced in [Soler2021]_ and available
+in Harmonica through the :class:`harmonica.EquivalentSourcesGB` class. The
+gradient-boosted equivalent sources divide the survey region in overlapping
+windows of equal size and fit the source coefficients iteratively, considering
+the sources and data points that fall under each window at a time. The order in
+which the windows are visited is randomized to improve convergence of the
+algorithm.
+
+Here we will produce a grid out of a portion of the ground gravity survey from
+South Africa (see :func:`harmonica.datasets.fetch_south_africa_gravity`) using
+the gradient-boosted equivalent sources. This particlar dataset is not very
+large, in fact we could use the :class:`harmonica.EquivalentSources` instead.
+But we will use the :class:`harmonica.EquivalentSourcesGB` for illustrating how
+to use them on a simple example.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import pyproj
+import boule as bl
+import verde as vd
+import harmonica as hm
+
+
+# Fetch the sample gravity data from South Africa
+data = hm.datasets.fetch_south_africa_gravity()
+
+# Slice a smaller portion of the survey data to speed-up calculations for this
+# example
+region = [18, 27, -34.5, -27]
+inside = vd.inside((data.longitude, data.latitude), region)
+data = data[inside]
+print("Number of data points:", data.shape[0])
+print("Mean height of observations:", data.elevation.mean())
+
+# Since this is a small area, we'll project our data and use Cartesian
+# coordinates
+projection = pyproj.Proj(proj="merc", lat_ts=data.latitude.mean())
+easting, northing = projection(data.longitude.values, data.latitude.values)
+coordinates = (easting, northing, data.elevation)
+
+# Compute the gravity disturbance
+ellipsoid = bl.WGS84
+data["gravity_disturbance"] = data.gravity - ellipsoid.normal_gravity(
+    data.latitude, data.elevation
+)
+
+# Start building the equivalent sources.
+# We'll use the block-averaged sources with a block size of 2km and windows of
+# 100km x 100km.
+window_size = 100e3
+block_size = 2e3
+
+# Let's estimate the memory required to store the largest Jacobian when using
+# these values for the window_size and the block_size.
+jacobian_req_memory = hm.EquivalentSourcesGB.estimate_required_memory(
```

Comment:
How about making this a regular method instead of a class method? Creating instances is cheap since the constructors don't do any computation. Then we could:

```python
eqs = hm.EquivalentSourcesGB(window_size=10, block_size=1)
print(eqs.estimate_required_memory(coordinates))
# Happy = carry on
# Too much = reduce window size and run again
```

This way you don't need to pass the arguments twice.

Reply:
Now that you put it in an example, it actually looks better as a regular method. I'll change it!